File: Unity/Example/retr0의 유니티 게임 프로그래밍 에센스/Hello Unity/Assets/Scripts/ch08/Calculator.cs
using UnityEngine;

public class Calculator : MonoBehaviour {
    private delegate float Calculate(float a, float b);
    private Calculate onCalculate;

    private float Sum(float a, float b) {
        Debug.Log(a + b);
        return a + b;
    }

    private float Sub(float a, float b) {
        Debug.Log(a - b);
        return a - b;
    }
    
    private float Mul(float a, float b) {
        Debug.Log(a * b);
        return a * b;
    }

    private void Start() {
        onCalculate = Sum;
        onCalculate += Sub;
        onCalculate -= Sub;
        onCalculate += Mul;
    }
    
    private void Update() {
        if(Input.GetKeyDown(KeyCode.Space)) {
            Debug.LogFormat("결과값: {0}", onCalculate(1, 10));
        }
    }
}

File: C++/Practice/Myeongpum Cpp Programming/Practice/prac3/Calculator.cpp
//
// Created by HM on 2021/08/20.
//

#include "Calculator.h"

void Practice_310::Add::setValue(int first, int second) {
     a = first; b = second;
}

int Practice_310::Add::calculate() {
    return a + b;
}

void Practice_310::Sub::setValue(int first, int second) {
    a = first; b = second;
}

int Practice_310::Sub::calculate() {
    return a - b;
}

void Practice_310::Mul::setValue(int first, int second) {
    a = first; b = second;
}

int Practice_310::Mul::calculate() {
    return a * b;
}

void Practice_310::Div::setValue(int first, int second) {
    a = first; b = second;
}

int Practice_310::Div::calculate() {
    return a / b;
}
